### PR TITLE
AP_Compass:   example test script didn't instantiate a serial_manager, and caused seg-fault

### PR DESCRIPTION
**Pull request by davidbuzz**
I also made the test so it doesn't run forever and block completion of the 'run.examples' target in the autotest.py

### CODE REVIEW

**Comment by davidbuzz** [comment]
@peterbarker  - for your review/pull.

**Comment by tridge** [comment]
doesn't build

File: libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp [hunk 1]
```diff
@@ -57,6 +60,8 @@ static void setup()
     timer = AP_HAL::micros();
 }
 
+int max_attempts = 200;
```
Comment (peterbarker) [inline libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp]: `uint8_t`

File: libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp [hunk 2]
```diff
@@ -78,7 +83,15 @@ static void loop()
 
             if (!compass.healthy()) {
                 hal.console->printf("not healthy\n");
-                continue;
+
+                max_attempts--;                 
+
+                // dont attempt at all for the first 100 iterations if not reporting healthy
```
Comment (peterbarker) [inline libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp]: This looks like it is tight-looping?  Perhaps a 100ms delay?

File: libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp [hunk 3]
```diff
@@ -78,7 +83,15 @@ static void loop()
...
             if (!compass.healthy()) {
                 hal.console->printf("not healthy\n");
-                continue;
+
+                max_attempts--;                 
+
+                // dont attempt at all for the first 100 iterations if not reporting healthy 
+                if ( max_attempts >= 100 ) { 
+                    continue; 
+                } else {
```
Comment (peterbarker) [inline libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp]: pointless else

File: libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp [hunk 4]
```diff
@@ -119,6 +132,9 @@ static void loop()
             hal.console->printf(" t=%u", (unsigned)read_time);
 
             hal.console->printf("\n");
+
+            // exit after 100-200 max attempts, that's enough to test it.
+            if ( max_attempts <= 0 ) { hal.scheduler->delay(100); exit(1); }
```
Comment (peterbarker) [inline libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp]: We do not exit in examples.

File: libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp (hunk 2 above)
Comment (davidbuzz) [reply libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp]: ... it's already in a 10Hz loop, so no it's not a 'tight loop'

File: libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp (hunk 4 above)
Comment (davidbuzz) [reply libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp]: really..?    I'm happy to change to a different construct for immediately giving-up if you point me at your prefered method ?

**Comment by davidbuzz** [comment]
peter and i diuscussed this and decided it's not necessary as the higher-level autotest tools have timeouts.

**Pull request closed without merging**